js/controller/foldercontroller.js

```diff
@@ -196,9 +196,11 @@ define(function(require) {
 
 		applyOnFolders(folders, function(folder) {
 			// Update total counter and prevent negative values
-			folder.set('total', Math.max(0, folder.get('total')));
+			folder.set('total', Math.max(0, folder.get('total')-1));
 		});
 
+		Radio.message.trigger('flag', message, 'unseen', false);
```

Comment:
Why is this line necessary? This will easily result in a race condition because the flag and delete requests are sent almost at the same time.

Reply:
Oh I see, yes this is not necessary. I'll remove it.